frontend/src/components/ShareButton.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useCallback, useState } from 'react';
+import { Button, Tooltip } from '@mui/material';
+
+export const ShareButton = ({ sharableText, ...rest }) => {
+	const [open, setOpen] = useState(false);
+
+	const share = useCallback(() => {
+		navigator.clipboard.writeText(sharableText);
+
+		setOpen(true);
+
+		setTimeout(() => {
+			setOpen(false);
+		}, 2000);
+	}, [sharableText]);
+
+	if (open) {
+		return (
+			<Tooltip title="Copied!" open>
```

Comment:
```suggestion
		return (
			<Tooltip title="Copied!" open={open}>
```

Always return the same structure